src/java/org/apache/cassandra/index/Index.java

```diff
@@ -426,6 +426,16 @@ public Indexer indexerFor(DecoratedKey key,
                               WriteContext ctx,
                               IndexTransaction.Type transactionType);
 
+    /**
+     * Tells whether this index supports replica fitering protection or not.
+     *
+     * @return true if this index supports replica filtering protection, false otherwise
+     */
+    default boolean supportsReplicaFilteringProtection()
```

Comment:
We could pass the `RowFilter` as an argument, so implementations can change their response depending on the specific query. For example, SASI could use RFP if no analyzed columns are involved. Not saying that we should change SASI behaviour in this ticket, just having the argument in the interface.